pyo3-macros-backend/src/method.rs

```diff
@@ -104,7 +106,7 @@ pub fn get_return_info(output: &syn::ReturnType) -> syn::Type {
     }
 }
 
-pub fn parse_method_receiver(arg: &syn::FnArg) -> syn::Result<SelfType> {
+pub fn parse_method_receiver(arg: &mut syn::FnArg) -> syn::Result<SelfType> {
```

Comment:
It doesn't look like `&mut` is needed here.

```suggestion
pub fn parse_method_receiver(arg: &syn::FnArg) -> syn::Result<SelfType> {
```

Reply:
I can't get rid of it yet because I still use `iter_mut()` to iterate over arguments:
https://github.com/PyO3/pyo3/blob/68137635d2eeecd8f8da610ca44a0c99806d91d8/pyo3-macros-backend/src/method.rs#L139-L145

I'll remove it once I figure out how to deal with this mutable/immutable arguments iterator